Short-Answer/Algorithms_Answers.md

```diff
@@ -1 +1,64 @@
-Add your answers to the Algorithms exercises here.
\ No newline at end of file
+Add your answers to the Algorithms exercises here.
+
+
+A ) the code would never end. the value of n would accelerate away from
+"a" linearly i believe. the while loop could be considered n^3 and the inner loop 
+its n^2+a which if subtracted reaches O(n) now I also have to consider "a" so maybe the time complexity is O(n + a)?  
+
+
+B) I think the time complexity of this function is O(n^4) 
+a normal double for loop is o(n^2) using this I cam to my conclusion 
+
+for i in range n,   length of n  
+        increase i 
+        for j in range i + 1, to n    
+        <!-- range is still determined by n but could be getting smaller because they increase the value of i by one -->
+            add one to j
+            for k in range k+ 1, to n      
+            <!-- range n is still being approached but this time k times k in being incrase by one for some reason -->
+                for l in range k + 1 to k + 10     
```

Comment:
So even though this a a nested `for...loop` it will only run a MAX of 10 times. No matter what `k` is, this loop will only run 10 times. You can plug in numbers to test. So even though this is a loop, it isn't linear in this case, this is `O(10)` which is equal to `O(1)` or constant time. So if we make that change this entire function's time complexity would be `O(n^3)` instead of `O(n^4)`.

Let me know if that explains it well. If not we can pair on it